### DISCUSSION TITLE
ECS subnet and windows server 2012/2016

### DISCUSSION THREAD

**Discussion by fatal-bundy** [PowerDNS/pdns#14786] [Q&A]
So we recently started sending ECS details via our anycast recursive servers. CDN caching reasons.
However we have come to find authoritative windows server DNS 2012/2016 does not like this EDNS option and sends a format error.

From what i can tell RFC wise all blame goes to microsoft but at that same token most of the big boy public resolvers seem to still handle this and do ECS.

i have some ideas how to handle it within the Powerdns ecosystem but wanted to see if the community had ran in to similar issues and if they had opinions on the matter

**Comment by omoerbeek** [comment]
In general yo want to send ECS info only to nameservers that handle it properly and base their replies on the ECS info received. So do not use a blanket https://docs.powerdns.com/recursor/settings.html#edns-subnet-allow-list, but make it specific to names that benefit from it.
